sys/net/gnrc/routing/rpl/gnrc_rpl.c

```diff
@@ -419,6 +419,18 @@ uint8_t gnrc_rpl_gen_instance_id(bool local)
     return instance_id;
 }
 
+void gnrc_rpl_configure_root(gnrc_netif_t *netif)
+{
+    gnrc_rpl_init(netif->pid);
+    gnrc_rpl_instance_t *inst = gnrc_rpl_instance_get(
+            CONFIG_GNRC_RPL_DEFAULT_INSTANCE
+        );
+    if (inst) {
+        gnrc_rpl_instance_remove(inst);
+    }
+    gnrc_rpl_root_init(CONFIG_GNRC_RPL_DEFAULT_INSTANCE, &addr, false, false);
```

Comment:
Where is `addr` coming from?

Reply:
I added `gnrc_netif_ipv6_get_global()` to get it

Reply:
Why not just provide it as a parameter? Guessing the address the DODAG should be opened on (is the one returned by `get_global()` really the one configured by the config protocol) seems like a bad idea. Not to mention that I do not see much use in such a guessing function in general.

Reply:
True and since `gnrc_netif_ipv6_addr_add_internal()` returns the index of the new address, we can just make use of that.